### PR TITLE
Enable IotLog override and fix format specifier warnings.

### PR DESCRIPTION
**Pull request by ravibhagavandas**
Fix Format specifier warnings
Allow logging override for log library.

Description
-----------

This PR contains changes enable overriding IotLog macro with a custom logging function by defining it in `iot_config.h`.
Also the PR aims to fix format specifier warnings like:

- Incorrect number of arguments to logger.
- Incorrect format specifier types

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c [hunk 1]
```diff
@@ -563,8 +563,7 @@ void _IotMqtt_DestroyOperation( _mqttOperation_t * pOperation )
         IotLogDebug( "(MQTT connection %p, %s operation %p) Removed operation from connection lists.",
                      pMqttConnection,
                      IotMqtt_OperationType( pOperation->u.operation.type ),
-                     pOperation,
-                     pMqttConnection );
+                     pOperation );
```
Comment (sarenameas) [inline libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c]: These probably needs to go the CSDK I will add this work item to my team. Thanks!

Comment (abhidixi11) [reply libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c]: It is already fixed in CSDK.

Comment (sarenameas) [reply libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c]: Oh nvm I saw the same problem on line 745.

**Pull request merged**